lib/services/template-service.ts

```diff
@@ -191,85 +109,85 @@ export class TemplateService implements ITemplateService {
         });
     }
 
-    public getAppTemplateDetails(templateName: string) {
-        let that = this,
-            templateDetails: any = {};
+    getAppTemplateDetails(templateName: string) {
+        const templateDetails: any = {};
 
-        return new Promise(function (resolve, reject) {
-            that.tmpPackageJsonFromSrc(templateName)
-                .then(function (pj: any) {
-                    let packageJson = pj;
-                    that.getTemplateMetaData(packageJson)
-                        .then(function (data: any) {
+        return new Promise((resolve, reject) => {
+            this.tmpPackageJsonFromSrc(templateName)
+                .then((packageJsonData: any) => {
+                    const packageJson = packageJsonData;
+                    this.getTemplateMetaData(packageJson)
+                        .then((data: any) => {
                             templateDetails.name = data.name;
                             templateDetails.displayName = data.displayName;
                             templateDetails.description = data.description;
                             templateDetails.version = data.version;
                             templateDetails.gitUrl = data.gitUrl;
                             templateDetails.type = data.type;
 
-                            return that.checkTemplateFlavor(packageJson);
+                            return this.checkTemplateFlavor(packageJson);
                         })
-                        .then(function (flav) {
-                            templateDetails.templateFlavor = flav;
-                            return that._getTmpAssetsContent(templateName);
+                        .then((flavor) => {
+                            templateDetails.templateFlavor = flavor;
+
+                            return this.getTmpAssetsContent(templateName);
                         })
-                        .then(function (resources) {
+                        .then((resources) => {
                             templateDetails.resources = resources;
                             resolve(templateDetails);
                         })
-                        .catch(function (error) {
+                        .catch((error) => {
                             reject({
-                                message: 'Error retrieving data for ' + templateName,
-                                error: error
+                                message: "Error retrieving data for " + templateName,
+                                error
                             });
                         });
                 })
-                .catch(function (error: any) {
+                .catch((error: any) => {
                     reject(error);
                 });
         });
     }
 
-    public getTemplates() {
-        let that = this,
-            tempDetails: Array<any> = [],
-            promises: Array<any> = [];
+    getTemplates() {
```

Comment:
"I would go much further with the refactoring, but I would like to talk with Bobi first"